src/dbnode/persist/fs/retriever_test.go

```diff
@@ -288,10 +361,52 @@ func testBlockRetrieverHighConcurrentSeeks(t *testing.T, shouldCacheShardIndices
 		}()
 	}
 
-	// Wait for all routines to be ready then start.
+	// Wait for all routines to be ready.
 	readyWg.Wait()
+	// Allow all the goroutines to begin.
 	startWg.Done()
 
+	// Setup concurrent block lease updates.
+	workers := xsync.NewWorkerPool(updateOpenLeaseConcurrency)
+	workers.Init()
+	// Volume -> shard -> blockStart to stripe as many shard/blockStart as quickly as possible to
+	// improve the chance of triggering the code path where UpdateOpenLease is the first time a set
+	// of seekers are opened for a shard/blocksStart combination.
+	for _, volume := range volumes {
+		for _, shard := range shards {
+			for _, blockStart := range blockStarts {
+				enqueueWg.Add(1)
+				var (
+					// Capture vars for async goroutine.
+					volume     = volume
+					shard      = shard
+					blockStart = blockStart
+				)
+				workers.Go(func() {
```

Comment:
nit: Is there any possibility this won't be done in order if spawn asynchronously?  Just a thought, if we think it's super unlikely to cause flakes/become an issue then I guess we can just ignore this possibility (without adding queueing/etc on a per volume/shard/blockstart tuple) and address it if it ever does happen.

Maybe at the least we should leave a note here about your thoughts on that?